lib/utils/exploits.ts

```diff
@@ -81,12 +83,44 @@ export const getUniqueChainIds = (exploit: Exploit): number[] => {
   return sortedChainIds;
 };
 
-export const isAffectedByExploit = (allowances: AllowanceData[], exploit: Exploit) => {
-  return allowances.some((allowance) => {
-    return exploit.addresses.some((address) => {
-      return address.chainId === allowance.chainId && address.address === allowance.spender;
+export const getExploitStatus = (allowances: AllowanceData[], exploit: Exploit) => {
+  const { events } = useAddressEvents();
```

Comment:
Hooks don't work inside functions, only in top-level functional components, so you'll need to move this to the `ExploitChecker` and pass the (parsed) events into the getExploitStatus function.